src/app/controllers/locations/create.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+flnd.locationCreate = {
+    configureSubmit: function($scope, config, messenger, authService, $uibModalInstance) {
+        return function() {
+            $scope.loading = true;
+
+            authService.post(config.locations, $scope.location)
+                .then(function success() {
```

Comment:
One improvement I want to make is to create "proxy" classes for different endpoints, so that the controllers merely call the proxies and deal with the results. Thus there would be a `LocationProxy` with an `create` method. The `LocationProxy` class would inject `authService`, which means that this `configureSubmit()` function would no longer depend on `authService`, making it that much cleaner.